token/program-2022/src/extension/mod.rs

```diff
@@ -160,26 +150,31 @@ fn type_and_tlv_indices<S: BaseState>(rest_input: &[u8]) -> Result<(usize, usize
 pub struct StateWithExtensions<'data, S: BaseState> {
     /// Unpacked base data
     pub base: S,
-    /// Unpacked account type
-    pub account_type: AccountType,
     /// Slice of data containing all TLV data, deserialized on demand
     tlv_data: &'data [u8],
 }
 impl<'data, S: BaseState> StateWithExtensions<'data, S> {
     /// Unpack base state, leaving the extension data as a slice
+    ///
+    /// Fails if the base state is not initialized.
     pub fn unpack(input: &'data [u8]) -> Result<Self, ProgramError> {
-        check_not_multisig(input)?;
+        check_min_len_and_not_multisig(input, S::LEN)?;
         let (base_data, rest) = input.split_at(S::LEN);
         let base = S::unpack(base_data)?;
-        let (type_index, tlv_start_index) = type_and_tlv_indices::<S>(rest)?;
-        let account_type = AccountType::try_from(rest[type_index])
-            .map_err(|_| ProgramError::InvalidAccountData)?;
-        check_account_type::<S>(account_type)?;
-        Ok(Self {
-            base,
-            account_type,
-            tlv_data: &rest[tlv_start_index..],
-        })
+        if let Some((account_type_index, tlv_start_index)) = type_and_tlv_indices::<S>(rest)? {
+            let account_type = AccountType::try_from(rest[account_type_index])
+                .map_err(|_| ProgramError::InvalidAccountData)?;
+            check_account_type::<S>(account_type)?;
+            Ok(Self {
+                base,
+                tlv_data: &rest[tlv_start_index..],
+            })
+        } else {
+            Ok(Self {
```

Comment:
A base mint/account is now handled

Reply:
Oops, definitely broke that case during a refactor at some point, nice catch